src/raftstore/store/fsm/peer.rs

```diff
@@ -582,10 +589,15 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
     }
 
     #[inline]
-    fn schedule_tick(&self, tick: PeerTick, timeout: Duration) {
+    fn schedule_tick(&mut self, tick: PeerTicks, timeout: Duration) {
+        if self.fsm.tick_registry.contains(tick) {
+            return;
+        }
         if is_zero_duration(&timeout) {
             return;
         }
+        trace!("schedule tick"; "tick" => ?tick, "timeout" => ?timeout, "region_id" => self.region_id(), "peer_id" => self.fsm.peer_id());
+        self.fsm.tick_registry.insert(tick);
```

Comment:
what if it failed to get mailbox or schedule peer tick, then the flag would be never removed

Reply:
The error can only happen when the node is shutting down. But it makes sense to keep registry clean.